### PR TITLE
formatting setup

### PR DESCRIPTION
**Pull request by chmelevskij**
* add pre-commit hooks to lint stage files
* prettier setup
* eslint setup and adjust current settings
* add recommended extension for prettier-eslint

There is a separate PR https://github.com/betaflight/betaflight-configurator/pull/3197 which actually applies all these changes. Keeping separate for clarity.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -20,7 +20,8 @@ cordova/bundle.keystore
 .DS_store
 
 # artefacts for Visual Studio Code
-/.vscode/
+.vscode/*
+!.vscode/extensions.json
```
Comment (chmelevskij) [inline .gitignore:24]: @McGiverGim Adding this so that we can recommend some extensions as well as having the commit hooks.

Comment (McGiverGim) [reply .gitignore:24]: Ok, I'm not too sure if this will work if the user has its own file.

File: .eslintrc.js [hunk 2]
```diff
@@ -12,13 +12,15 @@ module.exports = {
         webextensions: true,
     },
     rules: {
-        "no-trailing-spaces": "error",
-        "eol-last": "error",
```
Comment (McGiverGim) [inline .eslintrc.js:16]: Why are these removed? Maybe they are included in other part?

Comment (chmelevskij) [reply .eslintrc.js:16]: Exactly, formatter (pretier) is handling these kind of things better.

`linter` (eslint) is better for enforcing/discouraging specific patterns.

**Comment by chmelevskij** [comment]
Closing this as there seems to be no traction

**Comment by haslinghuis** [comment]
@chmelevskij sorry we simply did not get time for this.

**Comment by chmelevskij** [comment]
Lol, just saw you re-opened this 🫣

**Comment by haslinghuis** [comment]
Yes this would be helpful :crossed_fingers:

**Comment by haslinghuis** [comment]
But needs a rebase :see_no_evil:

**Comment by YarosMallorca** [comment]
This would be great! I would migrate to Prettier formatting.

**Review by blckmn — APPROVED** [review]

**Review by limonspb — APPROVED** [review]
